File: banchus-app/src/main/java/pe/nanamochi/banchus/controller/resource/MenuController.java
package pe.nanamochi.banchus.controller.resource;

import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import pe.nanamochi.banchus.domain.dto.MenuContentResponse;

@RestController
@RequestMapping("/")
@RequiredArgsConstructor
public class MenuController {
  @Value("${banchus.menu-icon.image-url}")
  private String menuIconImage;

  @Value("${banchus.menu-icon.redirect-url}")
  private String menuIconUrl;

  @GetMapping(value = "/menu-content.json")
  public MenuContentResponse getMenuContent() {
    MenuContentResponse.MenuImage icon =
        new MenuContentResponse.MenuImage(menuIconImage, menuIconUrl, true, null, null);
    return new MenuContentResponse(List.of(icon));
  }
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/controller/resource/MenuController.kt
package pe.nanamochi.banchus.controller.resource

import org.springframework.beans.factory.annotation.Value
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import pe.nanamochi.banchus.dto.MenuContentResponse

@RestController
@RequestMapping("/")
class MenuController(
    @Value("\${banchus.menu-icon.image-url}") private val menuIconImage: String,
    @Value("\${banchus.menu-icon.redirect-url}") private val menuIconUrl: String,
) {
    @GetMapping("/menu-content.json")
    fun getMenuContent() =
        MenuContentResponse(
            images = listOf(MenuContentResponse.MenuImage(image = menuIconImage, url = menuIconUrl))
        )
}
